pkg/controllers/framework/interface.go

```diff
@@ -81,6 +81,14 @@ func (s *Status) AsError() error {
 	return errors.New(s.Message())
 }
 
+// Error returns the err when Code is "Error" or "Misconfigured".
+func (s *Status) Error() error {
+	if s.IsError() {
```

Comment:
hrm, we might want to refactor here. why we have both isError and asError? couldn't we just check if returned error is nil?

Reply:
It's for the case when `Status` is nil I think. Remove this func. And modify the NewStatus(), to ensure AsError can always return s.err when code is Error or Misconfigured.